agent/taskresource/credentialspec/credentialspec_linux.go

```diff
@@ -0,0 +1,438 @@
+	defer wg.Done()
+	if iamCredentials == (credentials.IAMRoleCredentials{}) {
+		err := errors.New("credentialspec resource: unable to find execution role credentials")
+		cs.setTerminalReason(err.Error())
+		errorEvents <- err
+		return
+	}
+
+	_, err := arn.Parse(credentialspecS3ARN)
+	if err != nil {
+		cs.setTerminalReason(err.Error())
+		errorEvents <- err
+		return
+	}
+
+	bucket, key, err := s3.ParseS3ARN(credentialspecS3ARN)
+	if err != nil {
+		cs.setTerminalReason(err.Error())
+		errorEvents <- err
+		return
+	}
+
+	s3Client := cs.s3ClientCreator.NewS3Client(cs.region, iamCredentials)
+
+	credSpecJsonStringUnformatted, err := s3.GetObject(bucket, key, s3Client)
+
+	if err != nil {
+		cs.setTerminalReason(err.Error())
+		errorEvents <- err
+		return
+	}
+
+	credSpecJsonStringBytes := &bytes.Buffer{}
+	json.Compact(credSpecJsonStringBytes, []byte(credSpecJsonStringUnformatted))
+	credSpecJsonString := credSpecJsonStringBytes.String()
+
+	cs.updateCredSpecMapping(originalCredentialspec, credSpecJsonString)
+}
+
+func (cs *CredentialSpecResource) handleSSMCredentialspecFile(originalCredentialspec, credentialspecSSMARN string, iamCredentials credentials.IAMRoleCredentials, wg *sync.WaitGroup, errorEvents chan error) {
+	defer wg.Done()
+
+	if iamCredentials == (credentials.IAMRoleCredentials{}) {
+		err := errors.New("credentialspec resource: unable to find execution role credentials")
+		cs.setTerminalReason(err.Error())
+		errorEvents <- err
+		return
+	}
+
+	parsedARN, err := arn.Parse(credentialspecSSMARN)
+	if err != nil {
+		cs.setTerminalReason(err.Error())
+		errorEvents <- err
+		return
+	}
+
+	// An SSM ARN is in the form of arn:aws:ssm:us-west-2:123456789012:parameter/a/b. The parsed ARN value
+	// would be parameter/a/b. The following code gets the SSM parameter by passing "/a/b" value to the
+	// GetParametersFromSSM method to retrieve the value in the parameter.
+	ssmParam := strings.SplitAfterN(parsedARN.Resource, "parameter", 2)
+	if len(ssmParam) != 2 {
+		err := fmt.Errorf("the provided SSM parameter:%s is in an invalid format", parsedARN.Resource)
+		cs.setTerminalReason(err.Error())
+		errorEvents <- err
+		return
+	}
+	ssmParams := []string{ssmParam[1]}
+
+	ssmClient := cs.ssmClientCreator.NewSSMClient(cs.region, iamCredentials)
+	seelog.Errorf("ssm secret resource: retrieving resource for secrets %v in region [%s] in task: [%s]", cs.region, ssmParams)
+	ssmParamMap, err := ssm.GetSecretsFromSSM(ssmParams, ssmClient)
+	if err != nil {
+		errorEvents <- fmt.Errorf("fetching secret data from SSM Parameter Store in %s: %v", ssmParamMap, err)
+		return
+	}
+
+	ssmParamData := ssmParamMap[ssmParam[1]]
+	cs.updateCredSpecMapping(originalCredentialspec, ssmParamData)
+}
+
+func (cs *CredentialSpecResource) updateCredSpecMapping(credSpecInput, credSpecContent string) {
```

Comment:
please add a comment on what this func does

Reply:
added corresponding comment